ament_index_python/ament_index_python/__init__.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2015 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+
+RESOURCE_INDEX_SUBFOLDER = 'share/ament_index/resource_index'
+
+
+def get_search_paths():
+    """
+    Get the paths from the environment variable 'AMENT_PREFIX_PATH'.
+
+    :returns: list of paths
+    :raises: :exc:`EnvironmentError`
+    """
+    ament_prefix_path = os.environ.get('AMENT_PREFIX_PATH')
+    if not ament_prefix_path:
+        raise EnvironmentError("Environment variable 'AMENT_PREFIX_PATH' is not set or empty")
+
+    paths = ament_prefix_path.split(os.pathsep)
+    return [p for p in paths if p and os.path.exists(p)]
+
+
+def get_resource(resource_type, resource_name):
+    """
+    Get the content of a specific resource.
+
+    :param resource_type: the type of the resource
+    :type resource_type: str
+    :param resource_names: the name of the resource
+    :type resource_name: str
+    :returns: the content (bytes) of the resource
+    :raises: :exc:`EnvironmentError`
+    :raises: :exc:`LookupError`
+    """
+    assert resource_type, 'The resource type must not be empty'
+    assert resource_name, 'The resource type must not be empty'
+    for path in get_search_paths():
+        resource_path = os.path.join(path, RESOURCE_INDEX_SUBFOLDER, resource_type, resource_name)
+        if os.path.isfile(resource_path):
+            with open(resource_path, 'r') as h:
+                content = h.read()
+            return content
+    raise LookupError('Could not find the requested resource')
```

Comment:
Requiring the caller try except and reraise to insert the basic info about the exception is adding burden on a caller when the type and name are core to the exception. If I write a simple script without any saftey and I get a backtrace it should include the type and name as those are inherent to the error. If I call the method 3 times, in a loop I will be unable to detect what failed to lookup without adding a try except reraise.  If I'm adding try/except/reraise I would expect to add information which is only available in the calling context, and was not available when the error itself was generated.